d2/src/main/java/com/linkedin/d2/balancer/dualread/DualReadLoadBalancerJmx.java

```diff
@@ -29,6 +29,11 @@ public class DualReadLoadBalancerJmx implements DualReadLoadBalancerJmxMBean
   private final AtomicInteger _clusterPropertiesEvictCount = new AtomicInteger();
   private final AtomicInteger _uriPropertiesEvictCount = new AtomicInteger();
 
+  private final AtomicInteger _servicePropertiesOutOfSyncCount = new AtomicInteger();
```

Comment:
Don't seem to need any Atomic methods for these, should `volatile` be good enough?

Reply:
this is being written by ZK flow and indis flow, so we do need it.

Reply:
That's right, only `volatile` will not work since increment/decrement are composite operations. Here's a handy guide: https://www.baeldung.com/java-volatile-variables-thread-safety#:~:text=The%20volatile%20keyword%20is%20useful,that%20the%20operation%20is%20atomic.